pkgs/top-level/all-packages.nix

```diff
@@ -5260,16 +5260,18 @@ in
 
   rtags = callPackage ../development/tools/rtags/default.nix {};
 
+  rustc = rustcStable;
+  rustcStable = callPackage ../development/compilers/rustc/stable.nix {};
+  rustcBeta = callPackage ../development/compilers/rustc/beta.nix {};
   rustcMaster = lowPrio (callPackage ../development/compilers/rustc/head.nix {});
-  rustc = callPackage ../development/compilers/rustc {};
 
   rustPlatform = rustStable;
-
   rustStable = recurseIntoAttrs (makeRustPlatform cargo rustStable);
+  rustBeta = lowPrio (recurseIntoAttrs (makeRustPlatform cargoUnstable rustBeta));
   rustUnstable = lowPrio (recurseIntoAttrs (makeRustPlatform cargoUnstable rustUnstable));
 
   # rust platform to build cargo itself (with cargoSnapshot)
-  rustCargoPlatform = makeRustPlatform (cargoSnapshot rustc) rustCargoPlatform;
+  rustCargoPlatform = makeRustPlatform (cargoSnapshot rustcStable) rustCargoPlatform;
```

Comment:
We should go with `rustc` instead of `rustcStable` here. If users use `packageOverriders` to override `rustc`, they likely expect the other rust stuff to use the overriden `rustc` too